### PR TITLE
feat(s3): add new check `s3_multi_region_access_point_public_access_block`

### PR DESCRIPTION
**Pull request by HugoPBrito**
### Context

Amazon `S3 Multi-Region Access Points` facilitate data access across multiple `AWS regions` for globally distributed applications. By enabling `Block Public Access` settings for these `Access Points`, AWS helps prevent unauthorized public access to the data stored in `S3 buckets`. Without these settings, resources could be inadvertently exposed to the public internet, increasing risks of data breaches and unauthorized access.

### Description

This check ensures that `S3 Multi-Region Access Points` have `Block Public Access` settings enabled. If these settings are disabled, the check fails, indicating potential public exposure of data.

### Checklist

- Are there new checks included in this PR? Yes.
    - If so, do we need to update permissions for the provider? No.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/s3/s3_service.py [hunk 1]
```diff
@@ -525,6 +528,51 @@ def _list_access_points(self, regional_client):
...
+    def _list_multi_region_access_points(self):
+        logger.info("S3 - Listing account multi region access points...")
+        try:
+            list_multi_region_access_points = (
+                self.client.list_multi_region_access_points(
+                    AccountId=self.audited_account
+                )["AccessPoints"]
+            )
+            for mr_access_point in list_multi_region_access_points:
+                mr_ap_arn = f"arn:{self.audited_partition}:s3:{self.audited_account}:accesspoint/{mr_access_point['Name']}"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_service.py]:
```suggestion
                mr_ap_arn = f"arn:{self.audited_partition}:s3::{self.audited_account}:accesspoint/{mr_access_point['Name']}"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, make sure to always execute the checks in real infrastructure. This check does not work since: 
`
2024-10-29 09:28:00,360 [File: s3_service.py:568]       [Module: s3_service]     ERROR: us-east-1 -- ClientError[535]: An error occurred (PermanentRedirect) when calling the ListMultiRegionAccessPoints operation: This API operation is only available in the following Regions: us-west-2. Make sure to send all future requests to a supported Region.
`

Please, solve it.

**Comment by HugoPBrito** [comment]
> Please, make sure to always execute the checks in real infrastructure. This check does not work since: `2024-10-29 09:28:00,360 [File: s3_service.py:568] [Module: s3_service] ERROR: us-east-1 -- ClientError[535]: An error occurred (PermanentRedirect) when calling the ListMultiRegionAccessPoints operation: This API operation is only available in the following Regions: us-west-2. Make sure to send all future requests to a supported Region.`
> 
> Please, solve it.

I don't have the necessary permissions to test this on real infrastructure since the call is restricted to the `us-west-2` region. However, I was able to fix some errors. Beyond that, there's nothing more I can do without the required access.

At this point, there are no errors related to API call redirection. However, since I lack permissions, I’m unable to retrieve the data correctly. This results in an inaccurate setting for the `Public Access Block` on the `Multi-Region Access Point` in the infrastructure.

File: prowler/providers/aws/services/s3/s3_multi_region_access_point_public_access_block/s3_multi_region_access_point_public_access_block.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        findings = []
+        for mr_access_point in s3control_client.multi_region_access_points.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = mr_access_point.region
+            report.resource_id = mr_access_point.name
+            report.resource_arn = mr_access_point.arn
+            report.status = "PASS"
+            report.status_extended = f"S3 Multi Region Access Point {mr_access_point.name} of buckets {', '.join(mr_access_point.buckets)} does have Public Access Block enabled."
+
+            print(mr_access_point.public_access_block)
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/s3/s3_multi_region_access_point_public_access_block/s3_multi_region_access_point_public_access_block.py]:
```suggestion
```

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**